tests/test_codebase/test_mmdet/test_mmdet_models.py

```diff
@@ -2364,3 +2364,251 @@ def test_solov2_head_predict_by_feat(backend_type):
+        for i in range(5, 0, -1)
+    ]
+
+    # to get outputs of onnx/tensorrt model after rewrite
+    wrapped_model = WrapModel(
+        condinst_bbox_head, 'predict_by_feat', batch_img_metas=batch_img_metas)
+    rewrite_inputs = {
+        'cls_scores': cls_scores,
+        'bbox_preds': bbox_preds,
+        'score_factors': score_factors,
+        'param_preds': param_preds,
+    }
+    rewrite_outputs, is_backend_output = get_rewrite_outputs(
+        wrapped_model=wrapped_model,
+        model_inputs=rewrite_inputs,
+        deploy_cfg=deploy_cfg)
+
+    if is_backend_output:
+        dets = rewrite_outputs[0]
+        labels = rewrite_outputs[1]
+        param_preds = rewrite_outputs[2]
+        points = rewrite_outputs[3]
+        strides = rewrite_outputs[4]
+        assert dets.shape[-1] == 5
+        assert labels is not None
+        assert param_preds.shape[-1] == condinst_bbox_head.num_params
+        assert points.shape[-1] == 2
+        assert strides is not None
+    else:
+        assert rewrite_outputs is not None
+
+
+def get_condinst_mask_head():
+    """condinst Mask Head Config."""
+    test_cfg = Config(
+        dict(
+            mask_thr=0.5,
+            max_per_img=100,
+            min_bbox_size=0,
+            nms=dict(iou_threshold=0.6, type='nms'),
+            nms_pre=1000,
+            score_thr=0.05))
+    from mmdet.models.dense_heads import CondInstMaskHead
+    model = CondInstMaskHead(
+        mask_feature_head=dict(
+            end_level=2,
+            feat_channels=128,
+            in_channels=256,
+            mask_stride=8,
+            norm_cfg=dict(requires_grad=True, type='BN'),
+            num_stacked_convs=4,
+            out_channels=8,
+            start_level=0),
+        num_layers=3,
+        feat_channels=8,
+        mask_out_stride=4,
+        size_of_interest=8,
+        max_masks_to_train=300,
+        loss_mask=dict(
+            activate=True,
+            eps=5e-06,
+            loss_weight=1.0,
+            type='DiceLoss',
+            use_sigmoid=True),
+        test_cfg=test_cfg,
+    )
+
+    model.requires_grad_(False)
+    return model
+
+
+@pytest.mark.skipif(
+    reason='Only support GPU test', condition=not torch.cuda.is_available())
+@pytest.mark.parametrize('backend_type',
+                         [Backend.ONNXRUNTIME, Backend.TENSORRT])
+def test_condinst_mask_head_predict_by_feat(backend_type):
+    """Test predict_by_feat rewrite of condinst mask head."""
+    check_backend(backend_type)
+    s = 128
+    batch_img_metas = [{
+        'scale_factor': np.ones(4),
+        'pad_shape': (s, s, 3),
+        'img_shape': (s, s, 3)
+    }]
+
+    output_names = ['dets', 'labels', 'masks']
+    deploy_cfg = Config(
+        dict(
+            backend_config=dict(
+                type=backend_type.value,
+                common_config=dict(max_workspace_size=1 << 32),
+                model_inputs=[
```

Comment:
could remove these input shapes setting and they are not correct for inputs like:`mask_preds, det, label, batch_img_metas`